keps/sig-network/20200204-NetworkPolicy-verification-rearchitecture.md

```diff
@@ -0,0 +1,674 @@
+---
+title: Rearchitecting NetworkPolicy tests with a DSL for better upstream test coverage
+authors:
+  - "@jayunit100"
+  - "@abhiraut"
+  - "@sedefsavas"
+  - "@McCodeman"
+  - "@mattfenwick"
+owning-sig: sig-network
+reviewers:
+  - "bowei@"
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2020-02-04
+last-updated: 2020-5-01
+status: implementable
+---
+
+Note that this approach of higher level DSLs for testing may be moved into sig-testing for a broader set of tests over time.
```

Comment:
If we're going to put this into e2e.test (per the kubernetes/kubernetes PR), I think sig-testing should _at minimum_ be an interesting sig to begin with

Reply:
> I see no discussion of where this should live, or any of the logistics of using it.
> The approval process seems to have been perhaps a bit loose here ...

ah, ok, so the original KEP was to simple update the network_policy.go test, so there wasn't a real need for putting it anywhere new

Reply:
I updated this statement so that we don't cross streams - no intent of springing a new DSL on y'all, just some really necessary internal tooling for sig-net.  That said happy to present it at a sig-test mtng if y'all are interested

Reply:
... so on the subject of **how** this might be useful to sig-test... 

I think ginkgo test tables are going to be added in ginkgo v2, which might be worth talking about as a way to more broadly do table driven validation of various e2es..... https://docs.google.com/document/d/1h28ZknXRsTLPNNiOjdHIO-F2toCzq4xoZDXbfYaBdoQ/edit#

Reply:
I'm really excited about this KEP and associated PR because we're already basically doing this at my day job -- the truth tables approach of network policy verifications is something we get a lot of value out of.  It's important to us that we can clearly and easily verify that our cluster and network policies are set up and working as expected for security and auditing reasons.

If there's something in there that's KEP-worthy and goes beyond what this KEP covers, I'd be happy to build out a KEP from our use case!